astropy/units/tests/test_quantity_non_ufuncs.py

```diff
@@ -2752,7 +2753,8 @@ def test_known_arguments(self, target, helper):
             name for name in keyword_allowed_helper if not name.startswith("_")
         }
 
-        assert not (diff := keyword_allowed_helper - keyword_allowed_target), (
+        diff = keyword_allowed_helper - keyword_allowed_target
```

Comment:
Is this related? ruff has become anti-walrus? (I'm fine with it, just curious!)

Reply:
My interpretation is that @nstarman just manually rewrote those lines for readability, and I agree that walruses are hurtful in that regard when they're used to force 2 lines into a single one :)

Reply:
There is now a rule against assignments in `assert` statements: [RUF018](https://docs.astral.sh/ruff/rules/assignment-in-assert/). It is still allowed to use the walrus elsewhere.

Reply:
Hmm, that rule definitely does not apply here, the variable is very much not used elsewhere!

But anyway, this is fine!

Reply:
RUF018 forbids `assert ... := ...`, so it definitely does apply here.

Reply:
I meant that the intent of the rule doesn't apply here, `diff` is used only in the exception message.

Reply:
I separated the commit for RUF018 in this PR, so if a majority of folks are against it, it's easily removed. IMO it's largely been an improvement. Keeping `assert` statements very clean, with minimal extra points of failure is a +1. This feels kind of like being careful about what's in the `try` versus `else` parts of a `try-except-else`.

Reply:
I think we ended up all agreeing this was an improvement, so no worries!